2216/2/get_names.rb

```diff
@@ -0,0 +1,47 @@
+# script to get all names of participants of battles
+
+class ParticipantSelector
+  def self.extract_name(name)
+    participant = ''
+    if name.index('против')
+      participant = name[0..name.index('против') - 1]
+    elsif name.index('vs')
+      participant = name[0..name.index('vs') - 1]
+    elsif name.index('VS')
+      participant = name[0..name.index('VS') - 1]
+    end
+    participant.strip!
+  end
+
+  def remove_repeated_names(array_of_names, pairs)
+    pairs.each { |names| array_of_names.delete(find_deleted_name(names)) }
+  end
+
+  private
+
+  def find_deleted_name(names)
+    first_name = names[0]
+    second_name = names[1]
+    if first_name.size != second_name.size && (first_name.include?(second_name) || second_name.include?(first_name))
+      first_name.size > second_name.size ? first_name : second_name
+    else
+      ''
+    end
+  end
+end
+
+# ------------------------------------------------------------------------------
```

Comment:
I don't even want to try to understand what's going here.
The rule: **one file -- one class.** That's it.